exams/pearson/download.py

```diff
@@ -0,0 +1,248 @@
+"""Pearson SFTP download implementation"""
+import logging
+import os
+import zipfile
+from contextlib import contextmanager
+
+from django.conf import settings
+from paramiko import SSHException
+
+from exams.pearson.constants import (
+    EAC_SUCCESS_STATUS,
+    PEARSON_FILE_TYPE_EAC,
+    PEARSON_FILE_TYPE_VCDC,
+    VCDC_SUCCESS_STATUS,
+)
+from exams.pearson.exceptions import RetryableSFTPException
+from exams.pearson.readers import (
+    EACReader,
+    VCDCReader,
+)
+from exams.pearson import utils
+from exams.models import (
+    ExamAuthorization,
+    ExamProfile,
+)
+
+log = logging.getLogger(__name__)
+
+
+@contextmanager
+def locally_extracted(zip_file, member):
+    """
+    Context manager for temporarily extracting a zip file member to the local filesystem
+
+    Args:
+        zip_file (zipfile.ZipFile): the zip file to extract from
+        member (str|zipfile.ZipInfo): the name or ZipInfo of the member to extract
```

Comment:
Not in any current code, but the pysftp api accepts it. If you don't feel that's worth documenting I can simplify to just `str`.